lib/src/git.rs

```diff
@@ -322,15 +331,26 @@ pub fn export_refs(
     let mut branches_to_delete = BTreeMap::new();
     let mut failed_branches = vec![];
     let view = mut_repo.view();
-    let all_local_branch_names: HashSet<&str> = view
+    // This list includes refs known to jj, namely all git-tracking refs and refs
+    // for local jj branches.
+    // Short-term TODO: use the fact that git refs other than local branches are
+    // included.
+    let jj_known_refs: HashSet<_> = view
         .git_refs()
         .keys()
-        .filter_map(|r| ref_name_to_local_branch_name(r))
-        .chain(view.branches().keys().map(AsRef::as_ref))
+        .filter_map(|name| parse_git_ref(name))
+        .chain(
+            view.branches()
+                .keys()
+                .map(|branch| RefName::LocalBranch(branch.to_owned())),
+        )
         .collect();
-    for branch_name in all_local_branch_names {
-        let old_branch = view.get_git_ref(&local_branch_name_to_ref_name(branch_name));
-        let new_branch = view.get_local_branch(branch_name);
+    for jj_known_ref in jj_known_refs {
+        let (nice_branch_name, new_branch) = match &jj_known_ref {
+            RefName::LocalBranch(branch) => (branch.clone(), view.get_local_branch(branch)),
+            _ => continue,
+        };
+        let old_branch = view.get_git_ref(&to_git_ref_name(&jj_known_ref));
```

Comment:
Since `nice_branch_name` can be constructed from `parsed_ref_name`, we can track `failed_branches: Vec<RefName>` and convert them later. It will eliminate `nice_branch_name` from loop variables.

Reply:
Good point, thanks! I made `export_refs` return `Vec<RefName>`.